lib/record_store/zone/config/ignore_pattern.rb

```diff
@@ -0,0 +1,36 @@
+module RecordStore
+  class Zone
+    class Config
+      class IgnorePattern
+        attr_accessor :orig_hash
+        alias_method :to_hash, :orig_hash
+
+        def initialize(orig_hash)
+          @orig_hash = orig_hash
+        end
+
+        def should_ignore?(record)
+          all_pairs.all?{|(key, value)| record.respond_to?(key) && value_matches?(value, record.send(key))}
+        end
+
+        def all_pairs
+          orig_hash.reject{|key, value| key === 'match'}
+        end
+
+        def value_matches?(ignore_pattern_value, record_value)
+          return record_value.match?(ignore_pattern_value) if is_regex?
+          return record_value === ignore_pattern_value if is_exact?
+          false
+        end
+
+        def is_regex?
+          orig_hash['match'] === 'regex'
+        end
+
+        def is_exact?
```

Comment:
In ruby we usually don't use `is_<attr>` or `set_attr` or `get_attr`. Since ruby allows you to use special symbols as part of the method name, you can have `<attr>?`, `<attr>=` and `<attr>` as the equivalent version in Ruby.